packages/amplify-graphql-api-construct/src/amplify-graphql-api.ts

```diff
@@ -302,52 +297,35 @@ export class AmplifyGraphqlApi extends Construct {
     const deploymentType = this.node.tryGetContext(CDKContextKey.DEPLOYMENT_TYPE);
     const expectedDeploymentTypes = [this.SANDBOX_DEPLOYMENT_TYPE, this.BRANCH_DEPLOYMENT_TYPE];
 
-    // Throw an error if the deployment type is neither sandbox nor branch.
+    // Log a warning if the deployment type is neither sandbox nor branch.
     if (!expectedDeploymentTypes.includes(deploymentType)) {
-      throw new Error(
-        `${CDKContextKey.DEPLOYMENT_TYPE} must be one of (${expectedDeploymentTypes.join(', ')}). ` +
-        'Please set the deployment type by running:\n' +
-        'cdk --context amplify-backend-type=sandbox\n' +
-        'or\n' +
-        'cdk --context amplify-backend-type=branch'
+      console.warn(
+        `${CDKContextKey.DEPLOYMENT_TYPE} is recommended to be one of (${expectedDeploymentTypes.join(', ')}).\n` +
+          'Consider setting the deployment type by running:\n' +
+          'cdk --context amplify-backend-type=sandbox\n' +
+          'or\n' +
+          'cdk --context amplify-backend-type=branch\n',
       );
     }
```

Comment:
You should go ahead and return if we encounter this condition, since we can't safely proceed with an unknown deployment type.

Reply:
You should also return if we don't have a deployment type at all, right?

Reply:
> You should go ahead and return if we encounter this condition, since we can't safely proceed with an unknown deployment type.

If we return here on an unknown deployment type, we would not be checking namespace and name right?

Reply:
> You should also return if we don't have a deployment type at all, right?

Yeah i guess in that case returning an empty deploymentIdentifier would be appropriate